plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -307,46 +314,62 @@ public ConnectorPageSource createPageSource(
 
         Optional<PartitionData> partition = partitionSpec.isUnpartitioned() ? Optional.empty() : Optional.of(partitionData);
         LocationProvider locationProvider = getLocationProvider(table.getSchemaTableName(), table.getTableLocation(), table.getStorageProperties());
-        Supplier<IcebergPositionDeletePageSink> positionDeleteSink = () -> new IcebergPositionDeletePageSink(
-                split.getPath(),
-                partitionSpec,
-                partition,
-                locationProvider,
-                fileWriterFactory,
-                hdfsEnvironment,
-                hdfsContext,
-                jsonCodec,
-                session,
-                split.getFileFormat(),
-                table.getStorageProperties(),
-                split.getFileRecordCount());
-
-        Supplier<IcebergPageSink> updatedRowPageSinkSupplier = () -> new IcebergPageSink(
-                tableSchema,
-                PartitionSpecParser.fromJson(tableSchema, table.getPartitionSpecJson()),
-                locationProvider,
-                fileWriterFactory,
-                pageIndexerFactory,
-                hdfsEnvironment,
-                hdfsContext,
-                tableSchema.columns().stream().map(column -> getColumnHandle(column, typeManager)).collect(toImmutableList()),
-                jsonCodec,
-                session,
-                split.getFileFormat(),
-                table.getStorageProperties(),
-                maxOpenPartitions);
+        Supplier<ConnectorPageSink> positionDeleteSink = () -> {
+            try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
+                // Make sure we are using the PluginClassLoader when initialising and using the sink instance
+                return new ClassLoaderSafeConnectorPageSink(
```

Comment:
Do you need to wrap internal method calls or would just this be enough:
```
        Supplier<ConnectorPageSink> positionDeleteSink = () -> {
            try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
                // Make sure we are using the PluginClassLoader when initialising and using the sink instance
                return new IcebergPositionDeletePageSink(
                        split.getPath(),
                        partitionSpec,
                        partition,
                        locationProvider,
                        fileWriterFactory,
                        hdfsEnvironment,
                        hdfsContext,
                        jsonCodec,
                        session,
                        split.getFileFormat(),
                        table.getStorageProperties(),
                        split.getFileRecordCount());
            };
```

The alternative mathces what we do in `ClassLoaderSafeConnectorPageSourceProvider`

Reply:
I think it is enough to either do try/catch here or use `ClassLoaderSafeConnectorPageSink`

Reply:
try/catch is used for initialising the `IcebergPositionDeletePageSink`.
I thought that the `ClassLoaderSafeConnectorPageSink` wrapper is needed because the page sink is used within the methods of `UpdateablePageSource` (deleteRows/updateRows).
However, in the meantime I've wrapped the UpdateablePageSource with a class loader safe wrapper, so probably there is no need anymore to do the wrapping for the page sinks as well.